pkg/plugin/get.go

```diff
@@ -0,0 +1,180 @@
+  %[1]s get
+
+  # view DatadogAgent foo
+  %[1]s get foo
+`
+)
+
+// getOptions provides information required by Datadog get command
+type getOptions struct {
+	configFlags *genericclioptions.ConfigFlags
+	args        []string
+	client      client.Client
+	genericclioptions.IOStreams
+	userNamespace        string
+	userDatadogAgentName string
+}
+
+// newGetOptions provides an instance of getOptions with default values
+func newGetOptions(streams genericclioptions.IOStreams) *getOptions {
+	return &getOptions{
+		configFlags: genericclioptions.NewConfigFlags(false),
+		IOStreams:   streams,
+	}
+}
+
+// newCmdGet provides a cobra command wrapping getOptions
+func newCmdGet(streams genericclioptions.IOStreams) *cobra.Command {
+	o := newGetOptions(streams)
+	cmd := &cobra.Command{
+		Use:          "get [DatadogAgent name]",
+		Short:        "Get DatadogAgent deployment(s)",
+		Example:      fmt.Sprintf(getExample, "kubectl dd"),
+		SilenceUsage: true,
+		RunE: func(c *cobra.Command, args []string) error {
+			if err := o.complete(c, args); err != nil {
+				return err
+			}
+			if err := o.validate(); err != nil {
+				return err
+			}
+			return o.run()
+		},
+	}
+
+	o.configFlags.AddFlags(cmd.Flags())
+
+	return cmd
+}
+
+// complete sets all information required for processing the command
+func (o *getOptions) complete(cmd *cobra.Command, args []string) error {
+	o.args = args
+	var err error
+
+	clientConfig := o.configFlags.ToRawKubeConfigLoader()
+
+	// Create the Client for Read/Write operations.
+	o.client, err = newClient(clientConfig)
+	if err != nil {
+		return fmt.Errorf("unable to instantiate client: %v", err)
+	}
+
+	o.userNamespace, _, err = clientConfig.Namespace()
+	if err != nil {
+		return err
+	}
+
+	ns, err := cmd.Flags().GetString("namespace")
+	if err != nil {
+		return err
+	}
+
+	if ns != "" {
+		o.userNamespace = ns
+	}
+
+	if len(args) > 0 {
+		o.userDatadogAgentName = args[0]
+	}
+
+	return nil
+}
+
+// validate ensures that all required arguments and flag values are provided
+func (o *getOptions) validate() error {
+	if len(o.args) > 1 {
+		return fmt.Errorf("either one or no arguments are allowed")
+	}
+	return nil
+}
+
+// run runs the get command
+func (o *getOptions) run() error {
+	ddList := &v1alpha1.DatadogAgentList{}
+	if o.userDatadogAgentName == "" {
+		if err := o.client.List(context.TODO(), ddList, &client.ListOptions{Namespace: o.userNamespace}); err != nil {
+			return fmt.Errorf("unable to list DatadogAgent: %v", err)
+		}
+	} else {
+		dd := &v1alpha1.DatadogAgent{}
+		err := o.client.Get(context.TODO(), client.ObjectKey{Namespace: o.userNamespace, Name: o.userDatadogAgentName}, dd)
+		if err != nil && errors.IsNotFound(err) {
+			return fmt.Errorf("DatadogAgent %s/%s not found", o.userNamespace, o.userDatadogAgentName)
+		} else if err != nil {
+			return fmt.Errorf("unable to get DatadogAgent: %v", err)
+		}
+		ddList.Items = append(ddList.Items, *dd)
+	}
+
+	table := newGetTable(o.Out)
```

Comment:
coud you provide an example of the output.

I think the output should be equivalent to the `kubectl get dd`

```
$ k get dd
NAME            ACTIVE   AGENT             CLUSTER-AGENT         CLUSTER-CHECKS-RUNNER   AGE
datadog-agent   True     Running (3/3/3)   Progressing (2/2/2)   Progressing (2/2/2)     2d5h
```

Reply:
The current output can be reuse for another command: `kubectl dd describe datadog-agent`

Reply:
here is the current output of the command

```
  NAMESPACE  DATADOG NAME   AGE         COMPONENT      DESIRED  CURRENT  READY  UP-TO-DATE  AVAILABLE  STATUS
  default    datadog-agent  29 seconds  Agent          2        2        0      0           0          Updating (2/0/0)
  default    datadog-agent  29 seconds  Cluster Agent  1        0        0      1           0          Progressing (1/0/1)
```

It's similar to `kubectl get dd` but provides more explicit info (DESIRED, CURRENT, READY, AVAILABLE) about each component

Reply:
ok so it break a bit the usual "kubectl get" display.
imagine we have several `DatadogAgent` resource. it will be a bit complicated to follow.
WDYT?

but it make sens to have this information in: `kubectl datadog describe datadog-agent` (maybe in anther pr)

Reply:
makes sense 👍

Reply:
the new output
```
  NAMESPACE  NAME               AGENT            CLUSTER-AGENT     CLUSTER-CHECKS-RUNNER  AGE
  default    datadog-agent      Running (2/2/2)  Running (1/1/1)                          2 minutes
  default    datadog-agent-new  Running (2/2/2)  Updating (2/1/1)  Progressing (2/0/2)    39 seconds
```